pkgs/by-name/ri/rime-ice/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  unstableGitUpdater,
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "rime-ice";
+  version = "0-unstable-2024-09-16";
+
+  src = fetchFromGitHub {
+    owner = "iDvel";
+    repo = "rime-ice";
+    rev = "2ea99c5e4f3afaa24ebef61d17ea097557ca36b3";
+    hash = "sha256-AzH1rZFqEH8sovZZfJykvsEmCedEZWigQFHWHl6/PdE=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/rime-data
+    rm -rf ./others
+    cp -r ./* $out/share/rime-data
```

Comment:
I find that [some] [people] rename `default.yaml` to `rime_ice_suggestion.yaml` in their packaged `rime-ice`.  What are the pros and cons of doing so?  Should we do the renaming in Nixpkgs?  cc @linyinfeng @xddxdd 

---

@linyinfeng @xddxdd Would you like to become maintainers of this package in Nixpkgs?

[some]: https://github.com/xddxdd/nur-packages/blob/32a5830f36dfc3e6be5fe2f3f233138aaec36177/pkgs/uncategorized/rime-ice/default.nix#L13C21-L13C45
[people]: https://github.com/linyinfeng/nur-packages/blob/4ca82a926f92e9a640714c2ee1be1cc3585e4bb2/pkgs/rime-packages/rime-ice/default.nix#L104

Reply:
IMO overwriting `default.yaml` is bad, if users want to use multiple input schemas simultaneously, or switch between them. Users will effectively have to choose between `default.yaml` from one of the schemas. It should be renamed to a unique name to not conflict.

I'm OK with co-maintaining this package.

Reply:
I agree that name collision is bad.

I notice that there are more file name collisions between `rime-ice` and `rime-data`, such as `double_pinyin_flypy.schema.yaml`.  It seems much work is needed to avoid collisions completely.  Maybe `rime-ice` is not supposed to be used together with `rime-data`.

Reply:
> if users want to use multiple input schemas simultaneously, or switch between them. Users will effectively have to choose between default.yaml from one of the schemas.

How do you **choose** **one** `default.yaml` from many schemas?  If you mean something like this, it actually **merges** **two** yaml files: `default.yaml` and `rime_ice_suggestion.yaml`.

```yaml
# ~/.config/ibus/rime/default.custom.yaml
patch:
  __include: rime_ice_suggestion:/
```

Even if merging two files results in a working `default.yaml`, it is not a good user experience to have to write `__include: some_unknown_name:/` in `default.custom.yaml`.

---

IMHO, it is impossible to avoid file name collisions between schemas at distribution level.  It has to be done by users by setting `rimeDataPkgs` to a desired **ordered** list.

I suggest we merge this PR as is without any attempts to avoid collision.  WDYH?

Reply:
> IMHO, it is impossible to avoid file name collisions between schemas at distribution level. It has to be done by users by setting `rimeDataPkgs` to a desired **ordered** list.

It is possible, at least for `rime-ice`... The file name collision problem is caused by `rime-data` instead of `rime-ice`.

For example, in Arch Linux, instead of packaging a huge `rime-data` package, official rime data packages are packaged in a per-package manner.

`rime-ice` in AUR directly and indirectly depends on the following rime data packages:

1. `rime-bopomofo`
2. `rime-cangjie`
3. `rime-essay`
4. `rime-luna-pinyin`
5. `rime-prelude`
6. `rime-stroke`
7. `rime-terra-pinyin`

There is no file collision between all the Arch packages mentioned above. We have the file collision problem because `nixpkgs` packaged all official rime data packages into one big package `rime-data`, which includes `rime-double-pinyin` that conflicts with `rime-ice`.